File: lang/ruby/test/fun/vac/data_structures/LinkedStackTest.rb
require "fun/vac/data_structures/LinkedStack"
require "fun/vac/util/TestRunner"

def testLinkedStack()
    size = 8192

    stack = LinkedStack::LinkedStack.new()

    1.upto(size) do |i|
        stack.push(i)
    end

    if stack.size() != size
        return false
    end

    size.downto(1) do |i|
        if stack.peek() != i
            return false
        end
        stack.pop()
    end

    stack.isEmpty()
end

if __FILE__ == $PROGRAM_NAME
    TestRunner.parseTest(testLinkedStack())
end

File: lang/c/test/fun/vac/data_structures/LinkedStackTest.c
#include <fun/vac/data_structures/LinkedStack.h>
#include <fun/vac/util/TestRunner.h>

bool testLinkedStack(void) {
    size_t size = 8192;

    LinkedStack *stack = LinkedStack_new();

    for (size_t i = 1; i <= size; i++) {
        LinkedStack_push(stack, (int) i);
    }

    if (LinkedStack_size(stack) != size) {
        return false;
    }

    for (size_t i = size; i > 0; i--) {
        if ((size_t) LinkedStack_peek(stack) != i) {
            return false;
        }
        LinkedStack_pop(stack);
    }

    return LinkedStack_isEmpty(stack);
}

int main(void) {
    TestRunner_parseTest(testLinkedStack());
}
